core/go/internal/sequencer/coordinator/transaction/state_machine.go

```diff
@@ -0,0 +1,556 @@
+	Event_Assemble_Success                                                               // assemble response received from the originator
+	Event_Assemble_Revert_Response                                                       // assemble response received from the originator with a revert reason
+	Event_Endorsed                                                                       // endorsement received from one endorser
+	Event_EndorsedRejected                                                               // endorsement received from one endorser with a revert reason
+	Event_DependencyReady                                                                // another transaction, for which this transaction has a dependency on, has become ready for dispatch
+	Event_DependencyAssembled                                                            // another transaction, for which this transaction has a dependency on, has been assembled
+	Event_DependencyReverted                                                             // another transaction, for which this transaction has a dependency on, has been reverted
+	Event_DispatchRequestApproved                                                        // dispatch confirmation received from the originator
+	Event_DispatchRequestRejected                                                        // dispatch confirmation response received from the originator with a rejection
+	Event_Dispatched                                                                     // dispatched to the public TX manager
+	Event_Collected                                                                      // collected by the public TX manager
+	Event_NonceAllocated                                                                 // nonce allocated by the dispatcher thread
+	Event_Submitted                                                                      // submission made to the blockchain.  Each time this event is received, the submission hash is updated
+	Event_Confirmed                                                                      // confirmation received from the blockchain of either a successful or reverted transaction
+	Event_RequestTimeoutInterval                                                         // event emitted by the state machine on a regular period while we have pending requests
+	Event_StateTransition                                                                // event emitted by the state machine when a state transition occurs.  TODO should this be a separate enum?
+	Event_HeartbeatInterval                                                              // event emitted by the state machine on a regular period while we have pending requests
+	Event_AssembleTimeout                                                                // the assemble timeout period has passed since we sent the first assemble request
+)
+
+type StateMachine struct {
+	currentState    State
+	lastStateChange time.Time
+}
```

Comment:
Does seem like this should be a generic thing shared across state machines in the codebase.

That would allow over time:
- Enforcement of common naming/constructs
- Consistent logging for this critical system to diagnose
- RPC debug interfaces to query state machine state to be built once